scylla/src/transport/cluster.rs

```diff
@@ -517,6 +537,17 @@ impl ClusterWorker {
                         None => return, // If refresh_channel was closed then cluster was dropped, we can stop working
                     }
                 }
+                tablets_count = self.tablets_channel.recv_many(&mut tablets, 8192) => {
+                    if tablets_count == 0 {
+                        // If the channel was closed then cluster was dropped, we can stop working
+                        return;
+                    }
+                    let mut new_cluster_data: ClusterData = self.cluster_data.load().as_ref().clone();
```

Comment:
Afaik `load()` is preferable when we don't hold the object for a long time according to https://docs.rs/arc-swap/latest/arc_swap/struct.ArcSwapAny.html#method.load

Reply:
But for instance:
```rust
fn change_node_down_marker(&mut self, addr: SocketAddr, is_down: bool) {
    let cluster_data = self.cluster_data.load_full();
    // We need to iterate through the whole map here, but there will rarely more than ~100 nodes,
    // and changes of down marker are infrequent enough to afford this. As an important tradeoff,
    // we only keep one hashmap of known_peers, which is indexed by host IDs for node identification.
    let node = match cluster_data
        .known_peers
        .values()
        .find(|&peer| peer.address == NodeAddr::Translatable(addr))
    {
        Some(node) => node,
        None => {
            warn!("Unknown node address {}", addr);
            return;
        }
    };

    node.change_down_marker(is_down);
}
```
uses `load_full` even though it doesn't seem to hold the `cluster_data` for a long time.

Reply:
Are you asking me why the author of this code wrote it this way? You should ask the author, not me. I can only explain why I wrote my code the way I wrote it.
And I wrote my code this way because the `arc_swap` documentation says I should.

Reply:
I'm asking what are your thoughts after seeing this code. How do you explain the discrepancy? Do you believe that the code I gave uses `load_full` unnecessarily?

Reply:
Idk, maybe - you can open an issue about it if you want, as it's not related to this PR.